File: Greedy/1665.Minimum Initial Energy to Finish Tasks/Solution.java
class Solution {
    public int minimumEffort(int[][] tasks) {
        Arrays.sort(tasks, (a, b) -> (b[1] - b[0]) - (a[1] - a[0]));
        int ans = 0, left = 0;
        for (int[] t : tasks) {
            if (left >= t[1]) {
                left -= t[0];
            } else {
                ans += t[1] - left;
                left = t[1] - t[0];
            }
        }
        return ans;
    }
}

File: Greedy/1665.Minimum Initial Energy to Finish Tasks/Solution.py
class Solution:
    def minimumEffort(self, tasks: List[List[int]]) -> int:
        tasks.sort(key=lambda x:-(x[1] - x[0]))
        ans = left = 0
        for a, m in tasks:
            if left >= m:
                left -= a
            else:
                ans += m - left
                left = m - a
        return ans
